java/bogdana_maksiuta/src/com/zoolatech/lecture1/tasks/_7/OddNumbers.java

```diff
@@ -0,0 +1,57 @@
+package com.zoolatech.lecture1.tasks._7;
+
+import java.util.Scanner;
+
+/**
+ * 7. Write a program that prints all odd numbers from N to 1. N should be read from the input.
+ * Every number should be printed on the new line. Create two versions of the program:
+ * one should use the for loop, second - the while loop.
+ *         a. Input: 5
+ *            Output: 5
+ *                    3
+ *                    1
+ */
+public class OddNumbers {
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Enter a number: ");
+        int number = scanner.nextInt();
+
+        forLoopAllOddNumbers(number);
+        whileLoopAllOddNumbers(number);
+    }
+
+    public static void forLoopAllOddNumbers(int number) {
+        System.out.println("For loop: ");
+        if (number >= 0) {
+            for (int i = number; i > 0; i--) {
+                if (i % 2 == 1) {
+                    System.out.println(i);
+                }
+            }
+        } else {
+            for (int i = number; i < 1; i++) {
```

Comment:
Interesting, I haven't thought about this case. But, let's assume the number is positive.

Reply:
don't quite understand this condition. It's means that we don't need validation for the negative number?

Reply:
yes, we assume all numbers are positive